feat: completeopt support preinsert (#16403)
vim/vim: runtime/doc/options.txt

		    scores when "fuzzy" is enabled. Candidates will appear
		    in their original order.

	   preinsert

> 		    Preinsert the portion of the first candidate word that is
> 		    not part of the current completion leader and the
> 		    |hl-ComplMatchIns| highlight group is used,  Does not work
> 		    when fuzzy is also included.

Reviewer: ```suggestion
		    Preinsert the portion of the first candidate word that is
		    not part of the current completion leader using the
		    |hl-ComplMatchIns| highlight group.  Does not work with
		    "fuzzy".
```

> 		    |hl-ComplMatchIns| highlight group is used. Does not work
> 		    when "fuzzy" is also included.

Reviewer: ```suggestion
		    |hl-ComplMatchIns| highlight group. Does not work when
		    "fuzzy" is also included.
```
